lib/utilities.js

```diff
@@ -75,6 +76,20 @@ const buildOptions = function() {
     if (!options || typeof options !== 'object') return;
     Object.keys(options).forEach(i => result[i] = options[i]);
   });
+
+  // Ensure a hashAlgorithm option exists.
+  if (!result.hashAlgorithm) {
+    result.hashAlgorithm = 'md5';

```

Comment:
I would suggest to throw an error in this case, since we already have a default value and this probably a result of an error.

What do u think?

Reply:
To save you time, my current commit has your suggestion implemented. Feel free to ignore the below text if that satisfies you.

You did, however, ask for my thoughts. 0:-)

---

I went back and forth on this. 

Keep in mind, all of the text below is internal library design discussion. As written, all of these options work the same from an outside user's perspective.

## What I think `buildOptions` should be
In my opinion, the purpose of the `buildOptions` function should be to provide a valid Options object to be used by the rest of the application. It should be able to take in an empty object or no object, and still provide back a valid Options object with the minimal/required properties set. It should also do some kind of value validation before the return, as the user may have provided incorrect values or value types in their argument. It should provide the default values in the scope of the function, rather than requiring them to be passed in as the 1st argument to the function.

## What `buildOptions` currently is
Default values is not an intrinsic part of `buildOptions`. All it does currently is it creates a union of the provided objects, where overlapping values are overwritten by the last object in the arguments.

## What `buildOptions` is with my changes
The same as current, but if no argument provides hashAlgorithm, we return an object that has the default value. Developer can still pass in objects that don't contain a hashAlgorithm, and we just add it to the returned options.
We throw an error if the Developer provides an unsupported value for hashAlgorithm.

## What `buildOptions` is with the above suggestion
The same as current, but if no argument provides hashAlgorithm, we throw an error. The same behavior for if it receives an invalid hashAlgorithm.

I feel like the last one changes the intent of the function (as the current utilities tests describe it) as currently written, more than my previous implementation.